### PR TITLE
Sanity integration tests and Ray cluster test

### PR DESCRIPTION
**Pull request by sutaakar**
## Description
Implementing initial test coverage using integration tests based on Go.
Right now I cover just usage of Ray cluster, following PRs will cover also CodeFlare components.
First part of https://github.com/opendatahub-io/distributed-workloads/issues/63

## How Has This Been Tested?
By running the new Go tests against configured OCP with ODH deployment (based on quickstart instructions).

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by sutaakar** [comment]
Depends on https://github.com/opendatahub-io/distributed-workloads/pull/66

**Comment by sutaakar** [comment]
@winklerm FYI

File: tests/integration/ray_test.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+			},
+			ShutdownAfterJobFinishes: false,
+		},
+	}
+	rayJob, err = test.Client().Ray().RayV1alpha1().RayJobs(namespace.Name).Create(test.Ctx(), rayJob, metav1.CreateOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created RayJob %s/%s successfully", rayJob.Namespace, rayJob.Name)
+
+	// Retrieving the job logs once it has completed or timed out
+	defer support.WriteRayJobLogs(test, rayJob.Namespace, rayJob.Name)
```
Comment (astefanutti) [inline tests/integration/ray_test.go:220]: Maybe the `CODEFLARE_TEST_OUTPUT_DIR` environment variable can be mentioned in the README, that can be used to write test output in a non ephemeral directory.

File: tests/integration/sanity_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	support "github.com/project-codeflare/codeflare-operator/test/support"
```
Comment (astefanutti) [inline tests/integration/sanity_test.go:23]: nit: Move that import at the bottom of the import block.

File: tests/integration/ray_test.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+package integration
+
+import (
+	"encoding/base64"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	support "github.com/project-codeflare/codeflare-operator/test/support"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
```
Comment (astefanutti) [inline tests/integration/ray_test.go:25]: nit: Move those imports at the bottom of the import block.

**Review by astefanutti — APPROVED** [review]

**Pull request merged**